usaspending_api/awards/v2/data_layer/orm.py

```diff
@@ -455,3 +459,70 @@ def fetch_transaction_obligated_amount_by_internal_award_id(internal_award_id: i
         return _sum["transaction_obligated_amount__sum"]
 
     return None
+
+
+def fetch_psc_hierarchy(psc_code: str) -> dict:
+    if psc_code[0].isalpha():
+        codes = [psc_code, psc_code[:2], psc_code[:1]]
+        if psc_code[0] == "A":
+            codes = [psc_code, psc_code[:2], psc_code[:1], psc_code[:3]]
```

Comment:
Can we have a comment explaining this logic? This is some obscure stuff...

Reply:
In retrospect, I was just confused by listing the codes twice. Could we do:
```
codes = [psc_code, psc_code[:2], psc_code[:1], psc_code[:3] if psc_code[0] == "A" else None] 
```
instead? We could then skip the second == "A" check and just see if codes[4] is None instead, and I think make things more clear

Reply:
sure